integration-tests/tests/hacbs/add-component-to-application-via-hacbs-tab.spec.ts

```diff
@@ -45,44 +47,90 @@ describe('Create Components using the HACBS UI', () => {
             HACBSApplications.createdComponentExists(componentNames[1], applicationName);
         });
     });
 
     describe('Add a new component using the "Components" tab', () => {
-        it("Use HACBS 'Components' tabs to start adding a new compoent", () => {
-            HACBSApplications.goToComponentsTab();
-            componentsTabPage.clickAddComponent();
+        it("Use HACBS 'Components' tabs to start adding a new component", () => {
+            HACBSApplications.goToComponentsTab().clickAddComponent();
         });
 
         it('Add a component to Application', () => {
-            HACBSApplications.createComponent(publicRepos[1], componentNames[2]);
+            HACBSApplications.createComponent(publicRepos[1], componentNames[2], integrationTestNames[2]);
             HACBSApplications.createdComponentExists(componentNames[2], applicationName);
         });
     });
 
     describe('Add a new component using the "Actions" dropdown', () => {
         it("Click 'Actions' dropdown to add a component", () => {
-            cy.get(actionsDropdown.dropdown).click();
-            cy.contains('Add component').click();
+            HACBSApplications.clickActionsDropdown('Add component');
         });
 
         it('Add a component to Application', () => {
-            HACBSApplications.createComponent(publicRepos[2], componentNames[3]);
+            HACBSApplications.createComponent(publicRepos[2], componentNames[3], integrationTestNames[3]);
             HACBSApplications.createdComponentExists(componentNames[3], applicationName);
         });
     });
+
+    describe('Explore Integration Tests Tab', () => {
+        it("Click on 'Integration tests' tab and check the List View", () => {
+            HACBSApplications.goToIntegrationTestsTab();
+
+            for (let i = 0; i < 4; i++) {
+                integrationTestsTabPage.checkRowValues(integrationTestNames[i], containerImage, 'Mandatory', pipelineName);
+            }
+        });
+
+        it("Add a new Integration Test using 'Actions' dropdown, and mark it as Optional for release", () => {
+            HACBSApplications.clickActionsDropdown('Add integration test');
+            addIntegrationTestStep(integrationTestNames[4], true);
+
+            integrationTestsTabPage.checkRowValues(integrationTestNames[4], containerImage, 'Optional', pipelineName);
+        });
+
+        it("Filter Integration tests by 'Name'", () => {
+            // No Integration test should be visible
+            integrationTestsTabPage.filterByName('nothing');
+            cy.contains('No results found');
+
+            integrationTestsTabPage.filterByName('my-test');
+            // Only the first 4 Integration tests with prefix "my-test" should be visible
+            for (let i = 0; i < 4; i++) {
+                integrationTestsTabPage.checkRowValues(integrationTestNames[i], containerImage, 'Mandatory', pipelineName);
+            }
```

Comment:
use some mechanism that can iterate over a variable length, like a `for... of` loop